src/server/game/World/World.cpp

```diff
@@ -3293,6 +3289,14 @@ CharacterInfo const* World::GetCharacterInfo(ObjectGuid const& guid) const
     return nullptr;
 }
 
+ObjectGuid World::GetCharacterGuidByName(std::string const& name) const
```

Comment:
Since this is not using CharacterInfo, you can freely move it to its correct place, right?
Leave CharacterInfo for another PR.